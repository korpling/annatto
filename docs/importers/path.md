# path (importer)

Add file nodes for all files in the imported directory.

## Configuration

###  corpus_name



