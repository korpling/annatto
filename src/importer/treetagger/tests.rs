use std::path::Path;

use insta::assert_snapshot;

use crate::{importer::treetagger::TreeTaggerImporter, util::import_as_graphml_string};

const TT_DEFAULT_VIS_CONFIG: &str = r#"
[context]
default = 5
sizes = [0, 1, 2, 5, 10]

[view]
page_size = 10

[[visualizers]]
vis_type = "kwic"
display_name = "kwic"
visibility = "permanent"

[[visualizers]]
vis_type = "grid"
display_name = "grid"

[visualizers.mappings]
annos = "lb, highlighted"
"#;

#[test]
fn simple_token() {
    let actual = import_as_graphml_string(
        TreeTaggerImporter::default(),
        Path::new("tests/data/import/treetagger/token_only"),
        Some(TT_DEFAULT_VIS_CONFIG),
    )
    .unwrap();

    assert_snapshot!(actual);
}

#[test]
fn single_sentence() {
    let actual = import_as_graphml_string(
        TreeTaggerImporter::default(),
        Path::new("tests/data/import/treetagger/single_sentence"),
        Some(TT_DEFAULT_VIS_CONFIG),
    )
    .unwrap();

    assert_snapshot!(actual);
}
