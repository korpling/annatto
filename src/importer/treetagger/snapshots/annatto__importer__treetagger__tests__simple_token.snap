---
source: src/importer/treetagger/tests.rs
expression: actual
---
<?xml version="1.0" encoding="UTF-8"?>
<graphml>
    <key id="k0" for="graph" attr.name="configuration" attr.type="string"/>
    <key id="k1" for="node" attr.name="annis::doc" attr.type="string"/>
    <key id="k2" for="node" attr.name="annis::layer" attr.type="string"/>
    <key id="k3" for="node" attr.name="annis::node_type" attr.type="string"/>
    <key id="k4" for="node" attr.name="annis::tok" attr.type="string"/>
    <key id="k5" for="node" attr.name="annis::tok-whitespace-before" attr.type="string"/>
    <graph edgedefault="directed" parse.order="nodesfirst" parse.nodeids="free" parse.edgeids="canonical">
        <data key="k0"><![CDATA[
[context]
default = 5
sizes = [0, 1, 2, 5, 10]

[view]
page_size = 10

[[visualizers]]
vis_type = "kwic"
display_name = "kwic"
visibility = "permanent"

[[visualizers]]
vis_type = "grid"
display_name = "grid"
]]></data>
        <node id="token_only">
            <data key="k3">corpus</data>
        </node>
        <node id="token_only/zossen">
            <data key="k1">zossen</data>
            <data key="k3">corpus</data>
        </node>
        <node id="token_only/zossen#text">
            <data key="k3">datasource</data>
        </node>
        <node id="token_only/zossen#t1">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">Die</data>
        </node>
        <node id="token_only/zossen#t2">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">Jugendlichen</data>
            <data key="k5"> </data>
        </node>
        <node id="token_only/zossen#t3">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">in</data>
            <data key="k5"> </data>
        </node>
        <node id="token_only/zossen#t4">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">Zossen</data>
            <data key="k5"> </data>
        </node>
        <node id="token_only/zossen#t5">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">wollen</data>
            <data key="k5"> </data>
        </node>
        <node id="token_only/zossen#t6">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">ein</data>
            <data key="k5"> </data>
        </node>
        <node id="token_only/zossen#t7">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">Musikcaf√©</data>
            <data key="k5"> </data>
        </node>
        <node id="token_only/zossen#t8">
            <data key="k2">default_layer</data>
            <data key="k3">node</data>
            <data key="k4">.</data>
            <data key="k5"> </data>
        </node>
        <edge id="e0" source="token_only/zossen#t7" target="token_only/zossen#t8" label="Ordering/annis/">
        </edge>
        <edge id="e1" source="token_only/zossen#t6" target="token_only/zossen#t7" label="Ordering/annis/">
        </edge>
        <edge id="e2" source="token_only/zossen#t5" target="token_only/zossen#t6" label="Ordering/annis/">
        </edge>
        <edge id="e3" source="token_only/zossen#t4" target="token_only/zossen#t5" label="Ordering/annis/">
        </edge>
        <edge id="e4" source="token_only/zossen#t3" target="token_only/zossen#t4" label="Ordering/annis/">
        </edge>
        <edge id="e5" source="token_only/zossen#t2" target="token_only/zossen#t3" label="Ordering/annis/">
        </edge>
        <edge id="e6" source="token_only/zossen#t1" target="token_only/zossen#t2" label="Ordering/annis/">
        </edge>
        <edge id="e7" source="token_only/zossen" target="token_only" label="PartOf/annis/">
        </edge>
        <edge id="e8" source="token_only/zossen#text" target="token_only/zossen" label="PartOf/annis/">
        </edge>
        <edge id="e9" source="token_only/zossen#t1" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
        <edge id="e10" source="token_only/zossen#t2" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
        <edge id="e11" source="token_only/zossen#t3" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
        <edge id="e12" source="token_only/zossen#t4" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
        <edge id="e13" source="token_only/zossen#t5" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
        <edge id="e14" source="token_only/zossen#t6" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
        <edge id="e15" source="token_only/zossen#t7" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
        <edge id="e16" source="token_only/zossen#t8" target="token_only/zossen#text" label="PartOf/annis/">
        </edge>
    </graph>
</graphml>
